catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -422,10 +422,18 @@ public CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
           return queryResponse;
         };
 
-    List<Result> results =
-        ResultIterable.resultIterable(queryFunction, request, cqlRequest.getCount())
-            .stream()
-            .collect(Collectors.toList());
+    List<Result> results;
+    if (cqlRequest.getCount() == 0) {
+      results =
+          ResultIterable.resultIterable(queryFunction, request)
+              .stream()
+              .collect(Collectors.toList());
```

Comment:
❗️ Allowing someone to specify 0 and do an unlimited sized query creates a vulnerability because someone could just query an extremely large source and repeatedly run the jvm out of memory. 

We probably need some upper level max limit on this endpoint to prevent people from doing that.